lib/service/src/main/java/io/atlasmap/service/MappingService.java

```diff
@@ -204,6 +217,48 @@ public Response changeMappedFieldIndex(
         return Response.ok().build();
     }
 
+    /**
+     * Remove the specified field from the specified mapping in the mapping definition.
+     *
+     * @param mappingDefinitionId - mapping definition identifier (session)
+     * @param mappingId - mapping identifier within the specified session
+     * @param dataSourceType - source/ target
+     * @param fieldIndex - field index to be removed
+     *
+     * @return empty response
+     */
+    @DELETE
+    @Path("/{mappingId}/field/{dataSourceType}/{fieldIndex}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Remove Field From Mapping",
+        description = "Remove the specific field from the specified mapping in the mapping definition")
+    @ApiResponses({
+        @ApiResponse(responseCode = "200", description = "The mapping at the specified index within the specified mapping definition was removed successfully"),
+        @ApiResponse(responseCode = "204", description = "The specified mapping definition was not found")})
+    public Response removeFieldMappingRequest(
+        @Parameter(description = "Mapping Definition ID") @PathParam("mappingDefinitionId") Integer mappingDefinitionId,
+        @Parameter(description = "Mapping ID") @PathParam("mappingId") String mappingId,
+        @Parameter(description = "Source or Target Mapping") @PathParam("dataSourceType") DataSourceType dataSourceType,
+        @Parameter(description = "Field Index") @PathParam("fieldIndex") Integer fieldIndex) {
+        LOG.debug("removeFieldMappingRequest: ID: {}, mappingId: {}, dataSourceType: {}, fieldIndex: {}",
+            mappingDefinitionId, mappingId, dataSourceType, fieldIndex);
+        try {
+            ADMArchiveHandler handler = atlasService.getADMArchiveHandler(mappingDefinitionId);
+            AtlasMapping def = handler.getMappingDefinition();
```

Comment:
Can you create `AtlasMappingHandler#removeMappedField()` or something and move those logic to that? cf. https://github.com/atlasmap/atlasmap/issues/3988
you can get it through `ADMArchiveHandler#getAtlasMappingHandler()`